newbank/server/commands/MoveMoney.java

```diff
@@ -38,6 +43,17 @@ public CommandResponse process(ArrayList<String> argsList) throws CommandExcepti
                     String amountString = argsList.get(1);
                     double amount = Double.parseDouble(amountString);
 
+                    if (amount > 1000){
+                        if (theAuthenticator.containsKey(customer.getCustomerId())){
+                            String secretKey = theAuthenticator.getSecretKey(customer.getCustomerId());
+                            theAuthenticator.runAuthentication(secretKey);
+                            if (!string2FA.equals(theAuthenticator.code)){
+                                myResponse = "Incorrect 2FA Code";
+                                return new CommandResponse(myResponse);
+                            }
```

Comment:
Great stuff! 

I assume the logic of this piece of code can be similarly applied to withdrawing funds and transferring funds to other individuals.

Something we should chat about is if it's necessary to authenticate money moved from different accounts 'owned' by the same invidual.

Reply:
I would think it's unnecessary, but it's up to you. I do not want to squander the hard work you have put into this code.

Reply:
I agree, currently there are no commands for sending money to another customer or withdrawing money unless I am mistaken?